### ISSUE TITLE
shared memory reuse failed is softmax

### ISSUE THREAD

**Issue by liqiangxl** [NVIDIA/Fuser#1562]
repro:
```
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id2(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=DataType.Float, is_cpu=False)
    T1 = fd.ops.max(T0, axes=[1], keepdim=False, dtype=DataType.Null)
    S2 = fd.ops.size(T0, dim=0)
    S3 = fd.define_scalar(1, dtype=DataType.Int)
    V4 = fd.define_vector([S2, S3], dtype=DataType.Int)
    T5 = fd.ops.broadcast_in_dim(T1, shape=V4, broadcast_dims=[0])
    V6 = fd.ops.shape(T0)
    T7 = fd.ops.broadcast_in_dim(T5, shape=V6, broadcast_dims=[0, 1])
    T8 = fd.ops.sub(T0, T7)
    T9 = fd.ops.exp(T8)
    T10 = fd.ops.sum(T9, axes=[1], keepdim=False, dtype=DataType.Null)
    T11 = fd.ops.broadcast_in_dim(T10, shape=V4, broadcast_dims=[0])
    T12 = fd.ops.broadcast_in_dim(T11, shape=V6, broadcast_dims=[0, 1])
    T13 = fd.ops.reciprocal(T12)
    T14 = fd.ops.mul(T9, T13)
    fd.add_output(T14)

with FusionDefinition() as fd:
    nvfuser_fusion_id2(fd)

inputs = [
    torch.randn((5529600,), dtype=torch.float32, device='cuda:0').as_strided((160, 34560), (34560, 1)),
]
fd.execute(inputs)
```

**Comment by liqiangxl** [comment]
Shared memory is used to store persistent buffers in this size range. There are 2 persistent buffers in softmax, the 2nd one is supposed to reuse the shared memory of the 1st one. However, smem is not being reused due to two reasons: 
(1) there is an `expand` op in this python script, which is identified as incompatible with memory aliasing. This is a known issue https://github.com/NVIDIA/Fuser/issues/633
(2) For fp32, the access of the first buffer is vectorized but the 2nd buffer is not vectorized, this also disables the reuse.
